src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -286,32 +303,21 @@ impl ConcatSstableIterator {
                     ) != Ordering::Greater
                 })
             };
-
             while start_index < end_index {
-                let start_block_table_id = block_metas[start_index].table_id();
-                if table_info
-                    .get_table_ids()
-                    .binary_search(&start_block_table_id.table_id)
-                    .is_ok()
+                if self
+                    .existing_table_ids
+                    .contains(&block_metas[start_index].table_id().table_id)
                 {
                     break;
-                } else {
-                    start_index +=
-                        &block_metas[(start_index + 1)..].partition_point(|block_meta| {
-                            block_meta.table_id() == start_block_table_id
-                        }) + 1;
```

Comment:
Could you explain this change?

Reply:
Oh... I just think the count of block would not be much larger, so we can not filter them one by one....

Reply:
I have revert this change